File: fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/serializer/network/JsonChannelSerializer.java
package com.github.dedis.popstellar.model.network.serializer.network;

import com.github.dedis.popstellar.model.objects.Channel;
import com.google.gson.*;

import java.lang.reflect.Type;

public class JsonChannelSerializer implements JsonSerializer<Channel>, JsonDeserializer<Channel> {

  @Override
  public Channel deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context)
      throws JsonParseException {
    return Channel.fromString(json.getAsString());
  }

  @Override
  public JsonElement serialize(Channel channel, Type typeOfSrc, JsonSerializationContext context) {
    return new JsonPrimitive(channel.getAsString());
  }
}

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/serializer/network/JsonChannelSerializer.kt
package com.github.dedis.popstellar.model.network.serializer.network

import com.github.dedis.popstellar.model.objects.Channel
import com.google.gson.JsonDeserializationContext
import com.google.gson.JsonDeserializer
import com.google.gson.JsonElement
import com.google.gson.JsonParseException
import com.google.gson.JsonPrimitive
import com.google.gson.JsonSerializationContext
import com.google.gson.JsonSerializer
import java.lang.reflect.Type

class JsonChannelSerializer : JsonSerializer<Channel>, JsonDeserializer<Channel> {
    @Throws(JsonParseException::class)
    override fun deserialize(json: JsonElement, typeOfT: Type, context: JsonDeserializationContext): Channel {
        return Channel.fromString(json.asString)
    }

    override fun serialize(channel: Channel, typeOfSrc: Type, context: JsonSerializationContext): JsonElement {
        return JsonPrimitive(channel.asString)
    }
}